src/c1_foeppl_von_karman/foeppl_von_karman_curvable_bell_elements.h

```diff
@@ -46,672 +46,1160 @@
 
 
-    /// Interface to get the local indices of the nodes used by u
-    virtual Vector<unsigned> get_u_node_indices() const
-    {
-      return CurvableBellElement<NNODE_1D>::nodal_indices_for_field(
-        u_alpha_field_index(0));
-    }
 
-    /// Interface to get the local indices of the nodes used by w
-    virtual Vector<unsigned> get_w_node_indices() const
-    {
-      return CurvableBellElement<NNODE_1D>::nodal_indices_for_field(
-        w_field_index());
-    }
+  //============================================================================
+  /// FoepplVonKarmanC1CurvableBellElement elements are a subparametric scheme
+  /// with  linear Lagrange interpolation for approximating the geometry and
+  /// the C1-functions for approximating variables.
+  /// These elements use TElement<2,NNODE_1D> with Lagrangian bases to
+  /// interpolate the in-plane unknowns u_x, u_y although with sub-parametric
+  /// simplex shape interpolation.
+
+  /// The Bell Hermite basis is used to interpolate the out-of-plane unknown w
+  /// for straight sided elements and is upgraded to Bernadou basis if the
+  /// element is on a curved boundary. This upgrade similarly corrects the shape
+  /// interpolation.
+  ///
+  /// As a result, all nodes contain the two in-plane Lagrangian dofs, vertex
+  /// nodes (i.e. nodes 0,1,2) each contain an additional six out-of-plane
+  /// Hermite dofs.
+  ///
+  ///  e.g FeopplVonKarmanC1CurvableBellElement<4> (unupgraded)
+  ///                        | 0 | 1 | 2 | 3  | 4  |  5   |  6    |  7   |
+  ///          o <---------- |u_x|u_y| w |dwdx|dwdy|d2wdx2|d2wdxdy|d2wdy2|
+  ///         / \            .
+  ///        x   x <-------- |u_x|u_y|
+  ///       /     \          .
+  ///      x   x   x         .
+  ///     /         \        .
+  ///    o---x---x---o       .
+  ///
+  ///  e.g FeopplVonKarmanC1CurvableBellElement<3> (upgraded)
+  ///                         | 0 | 1 | 2 | 3  | 4  |  5   |  6    |  7   |
+  ///          o_ <---------- |u_x|u_y| w |dwdn|dwdt|d2wdn2|d2wdndt|d2wdt2|
+  ///         /  \            .
+  ///        / .  |           .
+  ///       x     x <-------- |u_x|u_y|
+  ///      / .  . |           .
+  ///     /        \          .
+  ///    o-----x----o         .
+  ///
+  //============================================================================
+
+  template<unsigned NNODE_1D>
+  class FoepplVonKarmanC1CurvableBellElement
+    : public virtual CurvableBellElement<NNODE_1D>,
+      public virtual FoepplVonKarmanEquations
+  {
+  public:
+
+    //----------------------------------------------------------------------
+    // Class construction
+
+    /// Constructor: Call constructors for C1CurvableBellElement and
+    /// FoepplVonKarmanEquations
+    FoepplVonKarmanC1CurvableBellElement()
+      : CurvableBellElement<NNODE_1D>(Nfield, Field_is_bell_interpolated),
+        FoepplVonKarmanEquations()
+    {
+
+      // hierher Aidan: Can this go; is this handled elsewhere?
```

Comment:
No, this still being commented out is a mistake from my debugging attempts. We should have a chat about which integration scheme is best for which version of the elements, though.

Reply:
Aidan: please tidy up brutally; I'll keep this comment in there to remind you but will mark this as resolved